src/backend/distributed/transaction/transaction_management.c

```diff
@@ -130,7 +130,16 @@ UseCoordinatedTransaction(void)
 
 	CurrentCoordinatedTransactionState = COORD_TRANS_STARTED;
 
-	AssignDistributedTransactionId();
+	/*
+	 * This might be part of bigger distributed transaction originating from
+	 * another node, in which case transaction id has already been assigned
+	 * by a assign_distributed_transaction_id() call.
```

Comment:
I wonder if we should have a new value for `CurrentCoordinatedTransactionState` to reflect that the current transaction is part of a coordinated transaction, but did not start it.

Reply:
> I wonder if we should have a new value for `CurrentCoordinatedTransactionState` to reflect that the current transaction is part of a coordinated transaction, but did not start it.

Added `COORD_TRANS_STARTED_ON_WORKER`.